### PR TITLE
#27 = Check if there's a tile before using it

### LINKED ISSUE #27: Cannot read properties of undefined (reading 'emit')
When running ``npm run start` and loading http://localhost:1234/ i get the following error:
```
RoomCamera.ts:198 Uncaught TypeError: Cannot read properties of undefined (reading 'emit')
    at HTMLCanvasElement._tilePointerMove (RoomCamera.ts:198:20)
```

image

### LINKED ISSUE #30: Random error when hovering furniture
It does not alwayss happen, but it happens a lot on moving furnis like doors
```
HitTexture.ts:93 Uncaught DOMException: Failed to execute 'getImageData' on 'CanvasRenderingContext2D': The source width is 0.
    at HitTexture._generateHitMap (http://localhost:1234/index.91126d5d.js:34670:35)
    at HitTexture._getHitMap (http://localhost:1234/index.91126d5d.js:34651:48)
    at HitTexture.hit (http://localhost:1234/index.91126d5d.js:34715:29)
    at FurnitureLayer.containsPoint (http://localhost:1234/index.91126d5d.js:34615:125)
    at EventBoundary.hitTestFn (http://localhost:1234/index.91126d5d.js:16363:63)
    at EventBoundary.hitTestRecursive (http://localhost:1234/index.91126d5d.js:16345:28)
    at EventBoundary.hitTestRecursive (http://localhost:1234/index.91126d5d.js:16337:40)
    at EventBoundary.hitTestRecursive (http://localhost:1234/index.91126d5d.js:16337:40)
    at EventBoundary.hitTestRecursive (http://localhost:1234/index.91126d5d.js:16337:40)
    at EventBoundary.hitTestRecursive (http://localhost:1234/index.91126d5d.js:16337:40)
```

### PR DESCRIPTION
**Pull request by tetreum**
Closes #27 & #30

### CODE REVIEW

File: src/objects/interactions/HitTexture.ts [hunk 1]
```diff
@@ -90,6 +90,10 @@ export class HitTexture {
         if (w > canvas.width) w = canvas.width;
         if (h > canvas.height) h = canvas.height;
 
+        if (w === 0) {
```
Comment (mathishouis) [inline src/objects/interactions/HitTexture.ts]: To improve readability and match the coding style, you could remove the brackets from the condition 😸 So we should have:
```js 
if (w === 0) return new Uint32Array();
```

File: src/objects/rooms/RoomCamera.ts [hunk 2]
```diff
@@ -172,7 +172,11 @@ export class RoomCamera extends Container {
      * @private
      */
     private _tilePointerDown = (event: PointerEvent): void => {
-        this._room.tiles.getTileFromGlobal({ x: event.clientX, y: event.clientY }).emit("pointerdown");
+        const tile = this._room.tiles.getTileFromGlobal({ x: event.clientX, y: event.clientY });
+
+        if (tile) { // may be hovering a black/empty space
```
Comment (mathishouis) [inline src/objects/rooms/RoomCamera.ts]: You can do the same here:
```js
if (tile) tile.emit("pointerdown");
```

File: src/objects/rooms/RoomCamera.ts [hunk 3]
```diff
@@ -182,7 +186,11 @@ export class RoomCamera extends Container {
      * @private
      */
     private _tilePointerUp = (event: PointerEvent): void => {
-        this._room.tiles.getTileFromGlobal({ x: event.clientX, y: event.clientY }).emit("pointerup");
+        const tile = this._room.tiles.getTileFromGlobal({ x: event.clientX, y: event.clientY });
+
+        if (tile) { // may be hovering a black/empty space
```
Comment (mathishouis) [inline src/objects/rooms/RoomCamera.ts]: And here:
```js
if (tile) tile.emit("pointerup");
```

File: src/objects/rooms/RoomCamera.ts [hunk 4]
```diff
@@ -193,6 +201,9 @@ export class RoomCamera extends Container {
      */
     private _tilePointerMove = (event: PointerEvent): void => {
         const object: Tile | Stair = this._room.tiles.getTileFromGlobal({ x: event.clientX, y: event.clientY });
+        if(object === undefined) {
```
Comment (mathishouis) [inline src/objects/rooms/RoomCamera.ts]: And also here:
```js
if(object === undefined) return;
```

**Review by mathishouis — CHANGES_REQUESTED** [review]
These issues are not really important, so if you don't have the time to fix that I can merge the pull request and fix that later 😸

**Comment by tetreum** [comment]
> These issues are not really important, so if you don't have the time to fix that I can merge the pull request and fix that later 😸

Maybe prettier should be installed to do this stuff automatically on pre-commit

**Review by mathishouis — APPROVED** [review]
Thanks!

**Pull request merged**